pkg/controller/deployment/deployment_controller.go

```diff
@@ -673,14 +675,7 @@ func lastRevision(allRSs []*extensions.ReplicaSet) int64 {
 // getOldReplicaSets returns two sets of old replica sets of the deployment. The first set of old replica sets doesn't include
 // the ones with no pods, and the second set of old replica sets include all old replica sets.
 func (dc *DeploymentController) getOldReplicaSets(deployment *extensions.Deployment) ([]*extensions.ReplicaSet, []*extensions.ReplicaSet, error) {
-	return deploymentutil.GetOldReplicaSetsFromLists(deployment, dc.client,
-		func(namespace string, options api.ListOptions) (*api.PodList, error) {
-			podList, err := dc.podStore.Pods(namespace).List(options.LabelSelector)
-			return &podList, err
-		},
-		func(namespace string, options api.ListOptions) ([]extensions.ReplicaSet, error) {
-			return dc.rsStore.ReplicaSets(namespace).List(options.LabelSelector)
-		})
+	return dc.getOldReplicaSetsAndSyncHashKey(deployment, dc.client)
```

Comment:
Why do you need to pass `dc.client` to a method of `dc`? It already has the client.